services/analyzer/sequelize-tables-analyzer.js

```diff
@@ -62,69 +62,140 @@ function hasTimestamps(fields) {
+    reference.isNotBelongsToMany = true;
+    reference.ref = foreignKey.foreignTableName;
+  } else if (association === 'belongsToMany') {
+    reference.isBelongsToMany = true;
+    reference.ref = manyToManyForeignKey.foreignTableName;
+    reference.otherKey = manyToManyForeignKey.columnName;
+    reference.junctionTable = foreignKey.tableName;
+  } else {
+    reference.isNotBelongsToMany = true;
+    reference.ref = foreignKey.tableName;
+  }
+
+  // NOTICE: If the foreign key name and alias are the same, Sequelize will crash, we need
+  //         to handle this specific scenario generating a different foreign key name.
+  if (reference.foreignKeyName && reference.foreignKeyName === reference.as) {
+    reference.foreignKeyName = `${reference.foreignKeyName}Key`;
+  }
+
+  if (foreignKey.foreignColumnName !== 'id') {
+    reference.targetKey = foreignKey.foreignColumnName;
+  }
+
+  return reference;
+}
+
 async function analyzeTable(table, config) {
   const schema = await analyzeFields(table, config);
 
   return {
     schema,
-    foreignKeys: await tableConstraintsGetter.perform(table),
+    constraints: await tableConstraintsGetter.perform(table),
     primaryKeys: await analyzePrimaryKeys(schema),
   };
 }
 
+// NOTICE: Use the foreign key and reference properties to determine the associations
+//         and push them as references of the table.
+function defineAssociationType(databaseSchema) {
+  Object.values(databaseSchema).forEach((table) => {
+    const { constraints, primaryKeys } = table;
+    constraints.forEach((constraint) => {
+      const { columnName } = constraint;
+      const uniqueIndexes = constraint.uniqueIndexes || null;
+
+      if (constraint.columnType === 'FOREIGN KEY') {
+        const isInCompositeKey = primaryKeys
+          && primaryKeys.length > 1
+          && primaryKeys.includes(columnName);
+
+        const { isPrimary, isUnique } = checkUnicity(primaryKeys, uniqueIndexes, columnName);
+
+        const refTableName = constraint.foreignTableName;
+        const refColumnName = constraint.foreignColumnName;
+        let isManyToMany = false;
+
+        if (isInCompositeKey) {
+          // Check if the foreignKey is in a composite primary key
+          primaryKeys.forEach((primaryKey) => {
+            if (primaryKey === constraint.columnName) {
+              const manyToManyKeys = _.filter(constraints, (otherKey) =>
+                otherKey.columnName !== constraint.columnName
+                  && otherKey.columnType === 'FOREIGN KEY' && primaryKeys.includes(otherKey.columnName)) || [];
+
+              manyToManyKeys.forEach((manyToManyKey) => {
+                databaseSchema[refTableName].references.push(createReference(constraint, 'belongsToMany', manyToManyKey));
+              });
+              isManyToMany = manyToManyKeys !== [];
+            }
+          });
+        }
+        if (!isManyToMany) {
+          const refPrimaryKeys = databaseSchema[refTableName].primaryKeys;
```

Comment:
[Conventions] not fan of `ref...` variable names we tend to name our variables with complete words.